maps-db/src/main/resources/db/migration/V1.00.01__create_map_index_table.sql

```diff
@@ -0,0 +1,33 @@
+comment on database maps_db is 'Database used to store, track and serve map information';
+
+create table map (
+  id serial primary key,
+  name varchar(256) not null unique check(length(name) > 3),
+  date_created timestamptz not null default now(),
+  date_updated timestamptz not null default now()
+);
+
+create table category (
+  id serial primary key,
+  name varchar(32) unique not null check (length(name) > 2),
+  date_created timestamptz not null default now()
+);
+
+insert into category (name) values
+  ('BEST'),
+  ('GOOD'),
+  ('EXPERIMENTAL');
+
+
+create table map_version (
+  id serial primary key,
+  map_id integer not null references map(id),
+  version integer not null check (version > 0),
+  url varchar(256) not null check (url like 'http%'),
+  thumbnail varchar(256) not null check (thumbnail like 'http%'),
```

Comment:
I don't think Map Thumbnails are a requirement right now.
So the not null constraint might be wrong, I could be wrong though

Reply:
This will be a good thing to check, you could be right.

I wonder if there should be room to update the map thumbnail. Immutable DB rows is a good thing, perhaps a lookup table would be in order.

Reply:
I think I found one:
https://triplea-game.org/map/big-world/

The way the websites handles a missing image is by using a fallback image.

I assume by 'room to update the map thumbnail' you mean within the same version, i.e. in order to update the thumbnail a new version needs to be pushed right?
In this case I'm not quite sure, both approaches have their pros and cons.
If you mean in general I wouldn't agree.

Reply:
If the thumbnail only is updated, I was thinking we'd have a mutable row and it would be an update statement.
If we want immutable rows, then we need a second table that has the listing of thumbnails. Thinking about it a bit more, IMO mutable rows here is fine.

> The way the websites handles a missing image is by using a fallback image.

Reply:
> The way the websites handles a missing image is by using a fallback image.

I'm curious what kind of traffic and usage the 'maps' section is getting (?)

I'm not sure if we should continue maintaining it.. The download link is extra functionality, the browsing of maps is totally redundant to what we have in-game. I'd like to see evidence that 'maps' on the website is having an impact (I could see that being a good and proper thing if this were a web-app, in which case that is how you would look through maps to play them perhaps, but for us...)

Reply:
Removed the thumbnail non-null constraint and added some additional UK constraints:  16d148b1363198440df7913e476f71665c6a73e2